bears/general/IndentationBear.py

```diff
@@ -0,0 +1,216 @@
+from coalib.bears.LocalBear import LocalBear
+from bears.general.AnnotationBear import AnnotationBear
+from bears.general.AnnotationBear import starts_within_ranges
+from coalib.parsing.StringProcessing.Core import unescaped_search_for
+from coalib.bearlib.languages.LanguageDefinition import LanguageDefinition
+from coalib.results.SourceRange import SourceRange
+from coalib.results.Result import Result, RESULT_SEVERITY
+from coalib.results.AbsolutePosition import AbsolutePosition
+from coalib.results.Diff import Diff
+
+
+class IndentationBear(LocalBear):
+
+    def run(self,
+            filename,
+            file,
+            dependency_results: dict,
+            language: str,
+            language_family: str,
+            use_spaces: bool=True,
+            tab_width: int=4):
+        """
+        It is a generic indent bear, which looks for a start and end
+        indent, example: ``{ : }`` where "{" is the start indent and
+        "}" is the end indent.
+
+        It does not however support hanging indents or absolute indents of
+        any sort, also indents based on keywords are not supported yet.
+        for example:
+
+            if(something)
+            does not get indented
+
+        undergoes no change.
+
+        :param file:
+            File that needs to be checked in the form of a list of strings.
+        :param filename:
+            Name of the file that needs to be checked.
+        :param language_family:
+            Language family to be used for indentation.
+        :param language:
+            Language to be used for indentation.
+        :param use_spaces:
+            Insert spaces instead of tabs for indentation.
+        :param tab_width:
+            No. of spaces to insert for indentation.
+            Only Applicable if use_spaces is False
```

Comment:
period missing?

also:

- please put the arguments in the same order in that they appear in the function definition above (line 15 - 21).
- please mention the depencendy_results argument